### ISSUE TITLE
Invalid JSON when parsing include from url query

### ISSUE THREAD

**Issue by sjovanovic** [fortunejs/fortune-http#12]
When include query contains json like in 

```
?include=foo,{"fields":{"bar":true,"baz":true}}
```

The json parsing fails with `SyntaxError: Unexpected end of JSON input` because value is split by comma and JSON contains comma.

This is in function `attachQueries` of `initialize_context.js` in line
```
path.push(JSON.parse(parts[1]))
```
I suggest replacing that with 

```
path.push(JSON.parse(parts.slice(1).join(',')))
```

**Comment by gr0uch** [comment]
No, this is working as expected. You need to modify your query to encode the JSON using `encodeURIComponent`:

```
?include=foo,%7B%22fields%22%3A%7B%22bar%22%3Atrue%2C%22baz%22%3Atrue%7D%7D
```

This will ensure that it will work no matter what contents are in the JSON.

**Comment by sjovanovic** [comment]
It is not working as expected unless it is standard to urlencode just a part of the query parameter's value. It is not standard.

If you urlencode the whole parameter as you're supposed to, you get the same bug.

Not to mention that even if I encode part of the parameter's value like you said it still doesn't fix the bug. I suggest reproducing before resolving.

**Comment by gr0uch** [comment]
I did try to reproduce the issue, and "fixed" it by URI encoding the JSON part.

Keep in mind that there needs to be some delimiter for the JSON part, so an unencoded `,` seemed to be a reasonable choice. There's nothing wrong with that, see here: https://stackoverflow.com/questions/45686595/possible-side-effect-using-comma-in-querystring/45702126#45702126

`,` and `%2C` are not equivalent. There is also no "standard" way since the URI standard doesn’t define any delimiters for the query component.

**Comment by sjovanovic** [comment]
Ok, I'm just trying to point out a bug, don't have any issues with format of the query string value other than it is not documented which part to urlencode and which to leave as is.

The issue is that this approach currently doesn't work, i.e. the bug is still there.

let's say I formated the query string like this:

```
include=foo,%7B%22fields%22%3A%7B%22bar%22%3Atrue%2C%22baz%22%3Atrue%7D%7D')

```

By the time it arrives to the line I pointed out in the issue it is already urldecoded:

```
> decodeURIComponent('foo,%7B%22fields%22%3A%7B%22bar%22%3Atrue%2C%22baz%22%3Atrue%7D%7D')
"foo,{"fields":{"bar":true,"baz":true}}"
```

And again there's the multiple comma issue.

I didn't say this earlier but the `SyntaxError: Unexpected end of JSON input` error is suppressed, I saw it by wrapping `JSON.parse` in try/catch.
